spec/features/users/sign_up_spec.rb

```diff
@@ -23,6 +23,21 @@
     end
   end
 
+  context 'user cancels on the enter password screen' do
+    it 'deletes the user record and returns them to the home page' do
+      email = 'test@test.com'
+
+      visit sign_up_email_path
+
+      submit_form_with_valid_email
+      click_confirmation_link_in_email(email)
+      click_on t('links.cancel_account_creation')
+
+      expect(current_path).to eq root_path
+      expect(User.find_with_email(email)).to be_nil
```

Comment:
@monfresh Absolutely, and I wasn't claiming that, I added this test in this PR because the bug wasn't caught. I was just stating that unit tests for the delete method already exist.